lib/web/apiserver.go

```diff
@@ -2701,3 +2707,18 @@ func ssoSetWebSessionAndRedirectURL(w http.ResponseWriter, r *http.Request, resp
 
 	return nil
 }
+
+// isRoleMatchError checks if an error is the result of a failed role match during
+// SSO authorization.
+func isRoleMatchError(err error) bool {
+	// oidc or saml
+	if trace.IsAccessDenied(err) {
+		return strings.HasPrefix(err.Error(), "unable to map attributes to role") ||
```

Comment:
Should we do Contains checks here instead, in case some data is added prepended to the error before this is called?